src/Services/PatientService.php

```diff
@@ -38,44 +39,48 @@ public function __construct()
     }
 
     // make this a comprehensive validation
-    public function validate($patient)
+    public function validate($patient, $context, $id = null)
     {
         $validator = new Validator();
-
-        $validator->required('fname')->lengthBetween(2, 255);
-        $validator->required('lname')->lengthBetween(2, 255);
-        $validator->required('sex')->lengthBetween(4, 30);
-        $validator->required('DOB')->datetime('Y-m-d');
-
-        return $validator->validate($patient);
-    }
-
-    public function validateUpdate($pid, $patient)
-    {
-        $validator = new Validator();
-
-        $vPid = $this->validatePid($pid);
-        if ($vPid->isNotValid()) {
-            return $vPid;
-        }
-
-        $validator->optional('fname')->lengthBetween(2, 255);
-        $validator->optional('lname')->lengthBetween(2, 255);
-        $validator->optional('sex')->lengthBetween(4, 30);
-        if (key_exists('DOB', $patient)) {
-            $validator->required('DOB')->datetime('Y-m-d');
+        if ($id) {
+            $vPid = $this->validatePid($id);
+            if ($vPid->isNotValid()) {
+                return $vPid;
+            }
         }
-
-        return $validator->validate($patient);
+        
+        $validator->context('insert', function (Validator $context) {
+            $context->required('first_name')->lengthBetween(2, 30);
+            $context->required('fname')->lengthBetween(2, 255);
+            $context->required('lname')->lengthBetween(2, 255);
+            $context->required('sex')->lengthBetween(4, 30);
+            $context->required('DOB')->datetime('Y-m-d');
+        });
+        
+        $validator->context('update', function (Validator $context) {
```

Comment:
http://validator.particle-php.com/en/latest/contexts/
I wanted to apply the example given there as "Extended example of copying".
I tried but didn't work for me I think its bug in docs or in Validator.
Can you please help me ?

Reply:
Oops I think we can't use this it doesn't copy the validation just copy the context :(